python/src/_dynamic_memory_index.py

```diff
@@ -108,55 +183,46 @@ def __init__(
-        :param alpha:
-        :type alpha: float
-        :param num_threads:
-        :type num_threads: int
-        :param filter_complexity:
-        :type filter_complexity: int
-        :param num_frozen_points:
-        :type num_frozen_points: int
-        :param initial_search_complexity: The working scratch memory allocated is predicated off of
-            initial_search_complexity * search_threads. If a larger list_size * num_threads value is
-            ultimately provided by the individual action executed in `batch_query` than provided in this constructor,
-            the scratch space is extended. If a smaller list_size * num_threads is provided by the action than the
-            constructor, the pre-allocated scratch space is used as-is.
-        :type initial_search_complexity: int
-        :param search_threads: Should be set to the most common batch_query num_threads size. The working
-            scratch memory allocated is predicated off of initial_search_list_size * initial_search_threads. If a
-            larger list_size * num_threads value is ultimately provided by the individual action executed in
-            `batch_query` than provided in this constructor, the scratch space is extended. If a smaller
-            list_size * num_threads is provided by the action than the constructor, the pre-allocated scratch space
-            is used as-is.
-        :type search_threads: int
-        :param concurrent_consolidation:
-        :type concurrent_consolidation: bool
+        The `diskannpy.DynamicMemoryIndex` represents our python API into a mutable DiskANN memory index.
+
+        This constructor is used to create a new, empty index. If you wish to load a previously saved index from disk,
+        please use the `diskannpy.DynamicMemoryIndex.from_file` classmethod instead.
+
+        ### Parameters
+        - **distance_metric**: A `str`, strictly one of {"l2", "mips", "cosine"}. `l2` and `cosine` are supported for all 3
+          vector dtypes, but `mips` is only available for single precision floats.
+        - **vector_dtype**: One of {`np.float32`, `np.int8`, `np.uint8`}. The dtype of the vectors this index will
+          be storing.
+        - **dimensions**: The vector dimensionality of this index. All new vectors inserted must be the same
+          dimensionality.
+        - **max_vectors**: Capacity of the data store including space for future insertions
+        - **graph_degree**: Graph degree (a.k.a. `R`) is the maximum degree allowed for a node in the index's graph
+          structure. This degree will be pruned throughout the course of the index build, but it will never grow beyond
+          this value. Higher `graph_degree` values require longer index build times, but may result in an index showing
+          excellent recall and latency characteristics.
+        - **saturate_graph**: If True, the adjacency list of each node will be saturated with neighbors to have exactly
```

Comment:
most of these args would be set to default rather than requiring the user to interpret, right? Can we place the default value in the doc here?

Reply:
They're all being sourced from a constants file so that we specifically kept them in sync across the builders and indices. Referencing their default value in the comment is just going to risk an incongruity between our documented values and our actual values being used.  I don't think it's worth it when the html docs render like 

![image](https://github.com/microsoft/DiskANN/assets/899411/581ab576-65e1-4fae-8982-be9804ad102e)

And if you're reading the docs via the `help()` function or jump-to-symbol in an IDE, you're just as capable of jumping to the consts file too. It only really helps during PR, not when most people will be reading the documentation as part of the whole file.  

If you still think we need it, I can add it, but I think our chances of making it less confusing are balanced by the risk of even more confusion if we ever update a default out of sync with the C++ constants.